editor/find_in_files_scanner.cpp

```diff
@@ -0,0 +1,352 @@
+		String relative_file_path = file_path + "::" + match.key;
+		DefaultFindInFilesScanner::scan(file, relative_file_path, match.value.display_text, search_text, match_case, whole_words, matches, Size2i(match.value.script_start_line, match.value.line_idx - 1));
+
+		// Let's handle some edge cases!
+		if (matches.size() > matches_offset) {
+			// We have new matches
+
+			// Check if the first new match is on the first line, if so handle it separately
+			if (matches.get(matches_offset).line_number == 1) {
+				FindInFilesScanner::ScanMatch scan_match = matches.get(matches_offset);
+				if (scan_match.begin - SCRIPT_SOURCE_TAG_LEN == scan_match.end - SCRIPT_SOURCE_TAG_LEN - 1) {
+					// Edge case matching the double quote mark
+					matches.remove_at(matches_offset);
+				} else {
+					// Another edge case where we need to remove "script/source = " from the resulting line
+					String actual_line = scan_match.line.replace("script/source = \"", "");
+					scan_match.file_path = relative_file_path;
+					scan_match.begin = scan_match.begin - SCRIPT_SOURCE_TAG_LEN;
+					scan_match.end = scan_match.end - SCRIPT_SOURCE_TAG_LEN;
+					scan_match.line = actual_line;
+					matches.insert(matches_offset, scan_match);
+					matches.remove_at(matches_offset + 1);
+				}
+			}
+
+			// Unescape all matched lines, this is necessary because of the way builtin scripts are stored.
+			// They get escaped before being written into the file, so things like quotes gets encoded into \".
+			int i = 0;
+			for (auto &scan_match : matches.slice(matches_offset)) {
+				String unescaped_line = scan_match.line.c_unescape();
+				int diffs_found = 0;
+				for (int j = 0; j <= scan_match.end; ++j) {
+					if (scan_match.line[j] == unescaped_line[j - diffs_found]) {
+						continue;
+					}
+
+					diffs_found++;
+				}
+				scan_match.begin -= diffs_found;
+				scan_match.end -= diffs_found;
+				scan_match.line = unescaped_line;
+				matches.insert(matches_offset + i, scan_match);
+				matches.remove_at(matches_offset + i + 1);
+				i++;
+			}
+
+			matches_offset = matches.size();
+		}
+	}
+}
+
+void BuiltinScriptFindInFilesScanner::replace(Ref<FileAccess> file, String file_path, Vector<FindInFilesScanner::ScanLocation> &locations, bool match_case, bool whole_words, String search_text, String new_text) {
+	for (auto &match : _parse_tscn(file)) {
+		if ((file->get_path() + "::" + match.key) != file_path) {
+			continue;
+		}
+
+		file->seek(0);
+		// Account for the edge cases also found during scanning
+		for (auto &location : locations) {
+			if (location.line_number == 1) {
+				location.begin += SCRIPT_SOURCE_TAG_LEN;
+				location.end += SCRIPT_SOURCE_TAG_LEN;
+			}
+		}
+		DefaultFindInFilesScanner::replace(file, file->get_path(), locations, match_case, whole_words, search_text, new_text, match.value.script_start_line - 1);
+	}
+}
+
+HashMap<String, BuiltinScriptFindInFilesScanner::SubResource> BuiltinScriptFindInFilesScanner::_parse_tscn(Ref<FileAccess> file) {
```

Comment:
See, something like this could probably be used in that other place where you have removed the ad-hoc parser.

Reply:
Just to reiterate on my previous comment on the subject, the "ad-hoc parser" relied on the fact that the search string was found **in** the .tscn file, literally. Therefore, the ad-hoc parser being right before opening the built-in script was necessary to make sure the string matched within the .tscn actually belonged to a script and wasn't just part of a Node's name or something.

My approach is different in that the .tscn file is properly parsed from the beginning and the parsing resulting is directly used to feed the Find In Files Panel, which makes it impossible to have it show up in the Panel and not be valid.